test/main.test.js

```diff
@@ -129,4 +129,40 @@ describe('LoopBack Context', function() {
       ], done);
     });
   });
+  it('handles concurrent then() calls with when v3.7.7 promises & bind option',
+  function() {
+    var timeout = 25;
+    function runWithPushedValue(pushedValue, bind, expectToFail) {
+      return new Promise(function concurrentExecution(outerResolve, reject) {
+        LoopBackContext.runInContext(function pushToContext() {
+          var ctx = LoopBackContext.getCurrentContext({bind: bind});
+          expect(ctx).is.an('object');
+          ctx.set('test-key', pushedValue);
+          var whenPromise = whenV377.promise(function(resolve) {
+            setTimeout(resolve, timeout);
+          });
+          whenPromise.then(function pullFromContext() {
+            var pulledValue = ctx && ctx.get('test-key');
+            return pulledValue;
+          }).then(function verify(pulledValue) {
+            if (bind) {
+              expect(pulledValue).to.equal(pushedValue);
+            } else if (expectToFail) {
+              expect(pulledValue).to.not.equal(pushedValue);
+            }
+            outerResolve();
+          }).catch(reject);
+        });
+      });
+    }
+    return Promise.all([
+      runWithPushedValue('test-value-1', true),
+      runWithPushedValue('test-value-2', true),
+    ]).then(function() {
+      return Promise.all([
```

Comment:
Please create two tests - one using bind, the other not using bind. That way when one of them fails, the second will be still  run.

Both tests can share the same `runWithPushedValue` function, which should be defined *after* the tests - see http://loopback.io/doc/en/contrib/style-guide.html